base_station/gui/src/components/ESTask.vue

```diff
@@ -0,0 +1,398 @@
+          }
+        }
+      }
+
+      const clamp = function (num, min, max) {
+        return num <= min ? min : num >= max ? max : num
+      }
+
+      servosMessage['pan'] = clamp(servosMessage['pan'], -1, 1)
+      servosMessage['tilt'] = clamp(servosMessage['tilt'], -1, 1)
+      this.lastServosMessage = servosMessage
+
+      this.lcm_.publish('/camera_servos', servosMessage)
+    }, 100)
+  },
+
+  components: {
+    Cameras,
+    CommIndicator,
+    ArmControls,
+    DriveControls,
+    EncoderCounts,
+    IKControls,
+    PDBFuse,
+    DriveVelDataV
+  }
+}
+</script>
+
+<style scoped>
+  .wrapper {
+    display: grid;
+    grid-gap: 10px;
+    grid-template-columns: 1fr 1.5fr;
+    grid-template-rows: 60px auto auto auto auto auto auto;
+    grid-template-areas: "header header"
+                         "ik-gui cameras"
+                         "controls ik-controls"
+                         "encoder ik-controls"
+                         "drive ik-controls"
+                         "pdb drive-motor"
+                         "pdb drive-motor";
+    font-family: sans-serif;
+    height: auto;
+  }
+
+  .box {
+    border-radius: 5px;
+    padding: 10px;
+    border: 1px solid black;
+  }
+
+  .light-bg {
+    background-color: LightGrey;
+  }
+
+  img {
+    border: none;
+    border-radius: 0px;
+  }
+
+  .header {
+    grid-area: header;
+    display: flex;
+    align-items: center;
+  }
+
+  .header h1 {
+    margin-left: 5px;
+  }
+
+  .spacer {
+    flex-grow: 0.8;
+  }
+
+  .comms {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+  }
+
+  .comms * {
+    margin-top: 2px;
+    margin-bottom: 2px;
+  }
+
+  .helpscreen {
+    z-index: 1000000000;
+    display: block;
+    visibility: hidden;
+    background-color: black;
+    opacity: 0.8;
+    position: absolute;
+    left: 0px;
+    top: 0px;
+    width: 100%;
+    height: 100%;
+  }
+
+  .helpimages {
+    z-index: 1000000001;
+    visibility: hidden;
+    position: absolute;
+    left: 5%;
+    top: 5%;
+    width: 90%;
+    height: 90%;
+  }
+
+  .help {
+    z-index: 1000000002;
+    display: flex;
+    float: right;
+    opacity: 0.8;
+    cursor: auto;
+  }
+
+  .help:hover {
+    opacity: 1.0;
+    cursor: pointer;
+  }
+
+  .help:hover ~ .helpscreen, .help:hover ~ .helpimages {
+    visibility: visible;
+  }
+
+  .diags {
+    grid-area: diags;
+  }
+
+  .ik-controls {
+    grid-area: ik-controls;
+  }
+
+  .controls {
+    grid-area: controls;
+    font-size: 1em;
+    display: flex;
+  }
+
+  .drive-motor {
+    grid-area: drive-motor;
+  }
+
+  .pdb {
+    grid-area: pdb;
+  }
+
+  .drive {
+    grid-area: drive;
+  }
+
+  .ik-gui {
+    grid-area: ik-gui;
+  }
+
+  .encoder {
+    grid-area: encoder;
+  }
+
+  .new-select {
+    display: inline-block;
+  }
+
+  .fil-hori-now {
+    margin-top: 20px;
+  }
+
+  .new-select {
+    display: inline-block;
+  }
+
+  .fil-hori-now {
+    margin-top: 20px;
+  }
+
+  ul#vitals li {
```

Comment:
is this correct syntax? it looks strange (not used to .vue so ignore if its fine)